__tests__/functional/tagged-format.test.ts

```diff
@@ -161,22 +164,51 @@ describe("tagged format", () => {
+    ${LONG_MIN}                    | ${LONG_MIN}                        | ${"bigint"}  | ${"@long"}   | ${"-(2**63)"}
+    ${Number.MIN_SAFE_INTEGER - 1} | ${Number.MIN_SAFE_INTEGER - 1}     | ${"number"}  | ${"@double"} | ${"-(2**53)"}
+    ${Number.MIN_SAFE_INTEGER}     | ${BigInt(Number.MIN_SAFE_INTEGER)} | ${"bigint"}  | ${"@long"}   | ${"-(2**53 - 1)"}
+    ${-(2 ** 31) - 1}              | ${BigInt(-(2 ** 31) - 1)}          | ${"bigint"}  | ${"@long"}   | ${"-(2**31) - 1"}
+    ${-(2 ** 31)}                  | ${-(2 ** 31)}                      | ${"number"}  | ${"@int"}    | ${"-(2**31)"}
+    ${0}                           | ${0}                               | ${"number"}  | ${"@int"}    | ${"0 (Int)"}
+    ${1}                           | ${1}                               | ${"number"}  | ${"@int"}    | ${"1 (Int)"}
+    ${BigInt("0")}                 | ${BigInt("0")}                     | ${"bigint"}  | ${"@long"}   | ${"0 (Long)"}
+    ${2 ** 31 - 1}                 | ${2 ** 31 - 1}                     | ${"number"}  | ${"@int"}    | ${"2**31 - 1"}
+    ${2 ** 31}                     | ${BigInt(2 ** 31)}                 | ${"bigint"}  | ${"@long"}   | ${"2**31"}
+    ${Number.MAX_SAFE_INTEGER}     | ${BigInt(Number.MAX_SAFE_INTEGER)} | ${"bigint"}  | ${"@long"}   | ${"2**53 - 1"}
+    ${Number.MAX_SAFE_INTEGER + 1} | ${Number.MAX_SAFE_INTEGER + 1}     | ${"number"}  | ${"@double"} | ${"2**53"}
+    ${LONG_MAX}                    | ${LONG_MAX}                        | ${"bigint"}  | ${"@long"}   | ${"2**64 - 1"}
+    ${1.3 ** 63}                   | ${1.3 ** 63}                       | ${"number"}  | ${"@double"} | ${"1.3**63"}
+    ${1.3}                         | ${1.3}                             | ${"number"}  | ${"@double"} | ${"1.3"}
   `(
     "Properly encodes and decodes number $testCase",
-    async ({ input, expected, expectedType, testCase }) => {
+    async ({ input, expected, expectedType, tag, testCase }) => {
       testCase;
+      const encoded = TaggedTypeFormat.encode(input);
+      const encodedKey = Object.keys(encoded)[0];
+      expect(encodedKey).toEqual(tag);
       const result = await client.query(fql`${input}`);
       expect(typeof result.data).toEqual(expectedType);
-      expect(result.data).toEqual(expected);
+      expect(result.data.toString()).toEqual(expected.toString());
     }
   );
+
+  it.each`
+    input                       | testCase
+    ${LONG_MIN - BigInt(1)}     | ${"lower than -(2**63) - 1"}
+    ${LONG_MAX + BigInt(1)}     | ${"greater than 2**63"}
+    ${Number.NEGATIVE_INFINITY} | ${"NEGATIVE_INFINITY"}
+    ${Number.POSITIVE_INFINITY} | ${"POSITIVE_INFINITY"}
+  `("Throws if BigInt value is $testCase", async ({ input }) => {
+    expect.assertions(2);
+    try {
+      const result = await client.query(fql`${input}`);
+      console.log(result);
+    } catch (e) {
+      if (e instanceof ClientError) {
```

Comment:
This should not be a client error - a client error implies a bug in our code.

Let's get some way to throw out an error here that clearly indicates to the user they sent us bad data.

(Perhaps we need to define a a basic IllegalArgumentError in the driver).

Let's cut a card for this for now.

Reply:
Using: https://faunadb.atlassian.net/browse/FE-2990